packages/web-components/fast-element/src/components/element-controller.ts

```diff
@@ -338,7 +341,10 @@ export class ElementController<TElement extends HTMLElement = HTMLElement>
 
         const behaviors = this.behaviors;
         if (behaviors !== null) {
-            for (const key of behaviors.keys()) {
+            // Iterate over the current attached behaviors only. Any behaviors
+            // added during this loop will be attached and connected in
+            // ElementController.addBehavior
+            for (const key of Array.from(behaviors.keys())) {
```

Comment:
I'm a little worried about creating the throw away array here. Is there another way we could handle this?

Reply:
I'm sure there is, I'll try to find an elegant approach. We need to guard against binding behaviors twice in the scenario when behaviors add other behaviors during connectedCallback

Reply:
Okay I changed this to cache the size of the map and then break out of the for loop after we connect a number of behaviors equal to that size. Because the `keys()` iterator returns items in insertion order, this will only bind the behaviors that are attached when the controller starts binding them.

Reply:
I think if we add a guard in addBehavior to not call connectedCallback when we're connecting them in connnectedCallback we can do this w/o odd behaviors or creating the array.